Remove dlib for performance improvements

DaveDuck321 opened the pull request:
In my tests, `cv2.dnn` is faster and more accurate than dlib (see `FaceDetectorDNN`). I've also added `FaceDetectorCascade` for faster but less accurate face detection.

Please check that `requirements.txt` won't break production

DaveDuck321 committed:
Remove dlib dependency for faster facial recognition

DLib and imutils were very slow for high resolution images and required
a large build on the initial installation. Fortunately, at this resolution,
`cv2.dnn` was ~4x faster, more accurate and already install with opencv.

Alternatively, I have provided `FaceDetectorCascade` which uses
`c2v.CascadeClassifier`. Although this is less accurate than dlib, I have
found that it is much faster at the required resolutions.

DaveDuck321 committed:
Makefile infrastructure for opencv face-detection

DaveDuck321 committed:
Revert to default port number

WeixuanZ reviewed (changes requested):
Face detection works great, but cropping seems to be a bit dodgy now with more accurate detection
@FChippendale could you have a look

DaveDuck321 commented:
> Face detection works great, but cropping seems to be a bit dodgy now with more accurate detection @FChippendale could you have a look

fyi I changed the defaults here: if it can't detect a face it'll set the entire image as a bounding box. We can revert this (with some other fix) if it messes with your assumptions in the cropping stage.

WeixuanZ committed:
Update Dockerfile

WeixuanZ committed:
Remove dlib from README

antroseco committed:
Improve containerization

- The Makefile runs yack in dev mode, because we want http locally.
- Dockerfile only copies required files to the container image.
- docker run called with --rm; there's no reason for the container to
  persist.

WeixuanZ reviewed (changes requested):
We shouldn’t copy secrets into the container - on production the env variables will be set in run config, while locally this will be explicitly passed into docker run command through Makefile variable substitution

Related to 9133f98e290c2977bb0f33dda3566eeda6933c0b

antroseco commented:
> We shouldn’t copy secrets into the container - on production the env variables will be set in run config, while locally this will be explicitly passed into docker run command through Makefile variable substitution
> 
> Related to [9133f98](https://github.com/WeixuanZ/hackcambridge2022/commit/9133f98e290c2977bb0f33dda3566eeda6933c0b)

Ah I missed that commit; let me rebase this branch on master to include it. What happens when `.secrets` doesn't exist though?

antroseco committed:
Self-sufficient Dockerfile

antroseco committed:
Make the Dockerfile slightly prettier

antroseco committed:
Follow Dockerfile best-practises

antroseco committed:
Conditionally load dotenv
